src/tails/node.ts

```diff
@@ -0,0 +1,229 @@
+// things from vite in that case or just reimplement the few extra fields?
+// Maybe it's the builder that needs to know this deep inheritance?
+const allNodeFrameworks: engine.Framework[] = [
+  {
+    name: "core:express",
+    parent: "nodejs",
+    dependencies: [{ name: "express" }],
+  },
+  {
+    name: "core:nextjs",
+    parent: "nodejs",
+    requiredFiles: [["next.config.js", "next.config.ts"]],
+    dependencies: [{ name: "next" }],
+  },
+  {
+    name: "core:angular",
+    parent: "nodejs",
+    dependencies: [{ name: "@angular/core" }, { name: "@angular/cli" }],
+    buildCommand: "ng build",
+    devCommand: "ng run",
+  },
+  {
+    name: "core:astro",
+    parent: "nodejs",
+    requiredFiles: [["astro.config.mjs", "astro.config.cjs", "astro.config.js", "astro.config.ts"]],
+    dependencies: [{ name: "astrojs" }],
+    canEmbed: ["core:svelte", "core:react"],
+    buildCommand: "astro build",
+    devCommand: "astro preview",
+  },
+  {
+    name: "core:react",
+    parent: "nodejs",
+    dependencies: [{ name: "react" }, { name: "react-dom" }],
+  },
+  {
+    name: "core:react-vite",
+    parent: "core:vite",
+    dependencies: [{ name: "react" }, { name: "react-dom" }],
+    vars: { vitePlugin: "react-jsx" },
+  },
+  {
+    name: "core:svelte",
+    parent: "nodejs",
+    // optionalFiles: svelte.config.js
+    dependencies: [{ name: "svelte" }],
+  },
+  {
+    name: "core:svelte-vite",
+    parent: "core:vite",
+    dependencies: [{ name: "svelte" }],
+    canEmbed: ["core:svelte"],
+    vars: { vitePlugin: "vite-plugin-svelte" },
+  },
+
+  // Note: Sveltekit is built on Vite but does not extend the vite note because
+  // the dependency is transitive
+  {
+    name: "core:sveltekit",
+    parent: "core:svelte",
+    dependencies: [{ name: "@sveltejs/kit" }],
+  },
+  {
+    name: "core:vite",
+    parent: "nodejs",
+    dependencies: [{ name: "vite" }],
+    buildCommand: "vite build",
+    devCommand: "vite preview",
+  },
+];
+
+// TODO: make this a JIT dynamic load from the filesystem
+
+type PackageManager = "npm" | "yarn";
+type Language = "js" | "ts-local" | "ts-global";
+interface PackageJson {
+  dependencies?: Record<string, string>;
+  devDependencies?: Record<string, string>;
+  scripts?: Record<string, unknown>;
+}
+
+export class Node implements engine.Strategy {
+  readonly runtimeName = "nodejs";
+  readonly frameworkName: string;
+  constructor(
+    private readonly pkgMgr: PackageManager,
+    private readonly lang: Language,
+    private readonly scripts: string[],
+    private readonly frameworks: engine.Framework[]
+  ) {
+    this.frameworkName = frameworks.length ? this.frameworks[0].name : "nodejs";
+  }
+
+  packageManagerInstallCommand(): string | null {
+    const packages: string[] = [];
+    if (this.pkgMgr === "yarn") {
+      packages.push("yarn");
+    }
+    if (this.lang === "ts-global") {
+      packages.push("typescript");
+    }
```

Comment:
What's the purpose of this? Wouldn't typesscript be part of `devDependencies`?

Reply:
```
npm install --global typescript
```

waaa didn't realize people actually do this!

Reply:
```
npm install --global typescript
```

Er, I'm one of those people that only does this. º_º